csharp_dotnetcore/12.NLP-With-LUIS/LuisBot/AppInsights/MyAppInsightLuisRecognizer.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.AI.Luis;
+using Newtonsoft.Json.Linq;
+
+namespace Luis_Bot.AppInsights
+{
+    /// <summary>
+    /// MyAppInsightLuisRecognizer invokes the Luis Recognizer and logs some results into Application Insights.
+    /// Logs the Top Intent, Sentiment (label/score), (Optionally) Original Text, Conversation and ActivityID.
+    /// The Custom Event name this logs is <see cref="MyLuisConstants.IntentPrefix"/> + "." + 'found intent name'
+    /// For example, if intent name was "add_calender":
+    ///    LuisIntent.add_calendar
+    /// See <see cref="LuisRecognizer"/> for additional information.
+    /// </summary>
```

Comment:
I'm trying to find a flaw with this comment.... but I can't! 

It's beautiful! 👍

Reply:
Thanks!